File: app/models/beer.rb
class Beer < ApplicationRecord
  has_many :user_beers
  has_many :users, through: :user_beers
  validates :name, presence: true
  validates :name, uniqueness: true
  validates :brewery, presence: true
  validates :style, presence: true
  scope :by_brewery, ->(brewery) {where(brewery: brewery)}

  def self.breweries
    self.all.collect{|x| x.brewery}.uniq
  end

  def display
    "#{self.name} - #{self.brewery} - #{self.style}"
  end

  def self.organize
    self.order(:brewery, :name)
  end

  def overallrating
    UserBeer.prep_rating(self).average(:rating)
  end

  def self.not_current_user_beers(user)
    @beers = []
    self.all.each do |beer|
      if user.beers.exclude?(beer)
        @beers << beer
      end
    end
    @beers
  end
end

File: app/assets/javascripts/beer.js
class Beer {
  constructor(id, name, brewery, style) {
    this.id = id
    this.name = name;
    this.brewery = brewery;
    this.style = style;
  }

 template() {
   return `<p><input type="checkbox" name="beer_ids[]" value="${this.id}">${this.name} - ${this.brewery} - ${this.style}</input> Rating: <input type="text" name="rating[${this.id}]"></p><br>`;
 }

 tableTemplate() {
   return `<tr><td>${this.name}</td><td>${this.brewery}</td><td>${this.style}</td></tr>`
 }
}
